### PR TITLE
Basic assertions

### PR DESCRIPTION
**Pull request by bogdanmarculescu**
As we discussed, basic assertions.
Checks are made against the content of the calls.
Added escapes for characters (for StringGene only) and a couple of tests for them.

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/search/gene/StringGene.kt [hunk 1]
```diff
@@ -30,8 +31,14 @@ class StringGene(
...
-    override fun getValueAsPrintableString(previousGenes: List<Gene>, mode: String?): String {
-        return "\"$value\""
+    override fun getValueAsPrintableString(previousGenes: List<Gene>, mode: String?, targetFormat: OutputFormat?): String {
+        when (targetFormat?.name){
+            OutputFormat.KOTLIN_JUNIT_5.name -> return "\"$value\""
+                                                            .replace("\\", "\\\\")
+                                                            .replace("$", "\\$")
+            else -> return return "\"$value\""
+                    .replace("\\", "\\\\")
+        }
```
[the remark this answers is not in the document]
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/search/gene/StringGene.kt]: Fixed.

File: core/src/main/kotlin/org/evomaster/core/search/gene/StringGene.kt [hunk 2]
```diff
@@ -30,8 +31,14 @@ class StringGene(
         value = randomness.nextWordString(minLength, Math.min(maxLength, maxForRandomizantion))
     }
 
-    override fun getValueAsPrintableString(previousGenes: List<Gene>, mode: String?): String {
-        return "\"$value\""
+    override fun getValueAsPrintableString(previousGenes: List<Gene>, mode: String?, targetFormat: OutputFormat?): String {
+        when (targetFormat?.name){
+            OutputFormat.KOTLIN_JUNIT_5.name -> return "\"$value\""
+                                                            .replace("\\", "\\\\")
```
[the remark this answers is not in the document]
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/search/gene/StringGene.kt]: Done.

File: core/src/main/kotlin/org/evomaster/core/search/gene/ObjectGene.kt [hunk 3]
```diff
@@ -48,7 +49,7 @@ open class ObjectGene(name: String, val fields: List<out Gene>, val refType : St
                 it !is CycleObjectGene &&
                         (it !is OptionalGene || it.isActive)
             }.map {
-                "\"${it.name}\":${it.getValueAsPrintableString(previousGenes,mode)}"
+                "\"${it.name}\":${it.getValueAsPrintableString(previousGenes, mode, null)}"
             }.joinTo(buffer, ", ")
```
[the remark this answers is not in the document]
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/search/gene/ObjectGene.kt]: I propagated targetFormat where applicable and added default behaviour (if targetFormat is null, just return the \"$value\" with no other escapes).

File: core/src/main/kotlin/org/evomaster/core/search/gene/Gene.kt [hunk 4]
```diff
@@ -61,11 +62,12 @@ abstract class Gene(var name: String) {
      */
     abstract fun getValueAsPrintableString(
             previousGenes: List<Gene> = listOf(),
-            mode: String? = null
+            mode: String? = null,
+            targetFormat: OutputFormat? = null
     ) : String
```
[the remark this answers is not in the document]
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/search/gene/Gene.kt]: Done.

File: core/src/main/kotlin/org/evomaster/core/search/gene/ArrayGene.kt [hunk 5]
```diff
@@ -56,9 +57,9 @@ class ArrayGene<T>(
         }
     }
 
-    override fun getValueAsPrintableString(previousGenes: List<Gene>, mode: String?): String {
+    override fun getValueAsPrintableString(previousGenes: List<Gene>, mode: String?, targetFormat: OutputFormat?): String {
         return "[" +
-                elements.map { g -> g.getValueAsPrintableString(previousGenes, mode) }.joinToString(", ") +
+                elements.map { g -> g.getValueAsPrintableString(previousGenes, mode, null) }.joinToString(", ") +
                 "]";
```
[the remark this answers is not in the document]
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/search/gene/ArrayGene.kt]: I propagated targetFormat where applicable and added default behaviour (if targetFormat is null, just return the "$value" with no other escapes).

File: core/src/main/kotlin/org/evomaster/core/output/service/TestSuiteWriter.kt [hunk 6]
```diff
@@ -134,8 +134,16 @@ class TestSuiteWriter {
         addImport(InsertionDto::class.qualifiedName!!, lines)
         addImport("java.util.List", lines)
         // TODO: BMR - this is temporarily added as WiP. Should we have a more targeted import (i.e. not import everything?)
+        if(config.enableBasicAssertions){
+            if (format.isJava()){
+                addImport("org.hamcrest.Matchers.*", lines, true)
+            }
+            else{
+                addImport("org.hamcrest.Matchers.*", lines, true)
+            }
```
[the remark this answers is not in the document]
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/output/service/TestSuiteWriter.kt]: Not sure what I was trying to do there. Fixed.

File: core/src/main/kotlin/org/evomaster/core/database/DbActionTransformer.kt [hunk 7]
```diff
@@ -56,10 +56,10 @@ object DbActionTransformer {
                     if (k is SqlForeignKeyGene) {
                         isFkReferenceToNonPrintable = handleSqlForeignKey(k, previous, entry)
                     } else {
-                        entry.printableValue = g.getValueAsPrintableString()
+                        entry.printableValue = g.getValueAsPrintableString(targetFormat = null)
                     }
```
[the remark this answers is not in the document]
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/database/DbActionTransformer.kt]: I propagated targetFormat where applicable and added default behaviour (if targetFormat is null, just return the "$value" with no other escapes).

File: core/src/main/kotlin/org/evomaster/core/output/TestCaseWriter.kt [hunk 8]
```diff
@@ -370,78 +371,54 @@ class TestCaseWriter {
...
                                     }
-
-                                    //lines.add(".body(\"${it}\", containsString(\"${resContents[it]}\"))")
-                                    //resContents[it]
                                 }
                             }
                             else -> {
                                 // this shouldn't be run if the JSON is okay. Panic! Update: could also be null. Pause, then panic!
                             }
                         }
```
[the remark this answers is not in the document]
Comment (bogdanmarculescu) [reply core/src/main/kotlin/org/evomaster/core/output/TestCaseWriter.kt]: Added basic handling (just compare strings) for this eventuality. We can add more detailed handling when needed.

File: core/src/main/kotlin/org/evomaster/core/search/gene/Gene.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (arcuri82) [reply core/src/main/kotlin/org/evomaster/core/search/gene/Gene.kt]: is it? It is not showing the diff

File: core/src/main/kotlin/org/evomaster/core/search/gene/StringGene.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (arcuri82) [reply core/src/main/kotlin/org/evomaster/core/search/gene/StringGene.kt]: is it? It is not showing the diff

**Pull request merged**